label_studio/tasks/models.py

```diff
@@ -997,6 +999,48 @@ def delete(self, *args, **kwargs):
         self.update_task()
         return result
 
+    @classmethod
+    def create_no_commit(cls, project, label_interface, task_id, data, model_version, model_run) -> Optional['Prediction']:
+        """
+        Creates a Prediction object from the given result data, without committing it to the database.
+        or returns None if it fails.
+
+        Args:
+            project: The Project object to associate with the Prediction object.
+            label_interface: The LabelInterface object to use to create regions from the result data.
+            data: The data to create the Prediction object with.
+                    Must contain keys that match control tags in labeling configuration
+                    Example:
+                        ```
+                        {
+                            'my_choices': 'positive',
+                            'my_textarea': 'generated document summary'
+                        }
+                        ```
+            task_id: The ID of the Task object to associate with the Prediction object.
+            model_version: The model version that produced the prediction.
+            model_run: The model run that created the prediction.
+        """
+        try:
+
+            # given the data receive, create annotation regions in LS format
+            # e.g. {"sentiment": "positive"} -> {"value": {"choices": ["positive"]}, "from_name": "", "to_name": "", ..}
+            pred = PredictionValue(result=label_interface.create_regions(data)).model_dump()
```

Comment:
nit: I figure it's slightly better to just use the PredictionValue here, and we can just do `pred.result` below rather than `pred['result']`?
Obviously not a bit deal, but generally it seems like we've been quick to `model_dump` things, and I'd like to just use the pydantic model where we can.